### PR TITLE
Chainparams: Refactor: Remove redundant AllowMinDifficultyBlocks() getter

### PR DESCRIPTION
**Pull request by jtimon**
A small part of #5970.

### CODE REVIEW

File: src/miner.h [hunk 1]
```diff
@@ -6,6 +6,7 @@
 #ifndef BITCOIN_MINER_H
 #define BITCOIN_MINER_H
 
+#include "consensus/params.h"
 #include "primitives/block.h"
```
Comment (theuni) [inline src/miner.h]: Forward-declare instead?

**Comment by theuni** [comment]
ut ACK other than the nit.

**Comment by sipa** [comment]
ut ACK

**Comment by jtimon** [comment]
nit (forward declaration of Consensus::Params) fixed

File: src/miner.h [hunk 2]
```diff
@@ -29,6 +30,6 @@ CBlockTemplate* CreateNewBlock(const CScript& scriptPubKeyIn);
 CBlockTemplate* CreateNewBlockWithKey(CReserveKey& reservekey);
 /** Modify the extranonce in a block */
 void IncrementExtraNonce(CBlock* pblock, CBlockIndex* pindexPrev, unsigned int& nExtraNonce);
-void UpdateTime(CBlockHeader* block, const CBlockIndex* pindexPrev);
+void UpdateTime(CBlockHeader*, const Consensus::Params&, const CBlockIndex*);
```
Comment (laanwj) [inline src/miner.h]: I don't agree on removing the argument names here, they serve as a kind of documentation.

**Comment by laanwj** [comment]
utACK apart from above nit

**Comment by jtimon** [comment]
Fixed @laanwj 's nit (do not remove the parameter names of the functions in the header file).

File: src/miner.h [hunk 3]
```diff
@@ -14,6 +14,7 @@ class CBlockIndex;
 class CReserveKey;
 class CScript;
 class CWallet;
+namespace Consensus { class Params; };
```
Comment (Diapolo) [inline src/miner.h]: Nit: If you mind add a new-line here, but that's not meant to be of any interrest if this is going to get merged now :).

**Pull request merged**